internal/querycoord/cluster.go

```diff
@@ -643,8 +668,12 @@ func (c *queryNodeCluster) removeNodeInfo(nodeID int64) error {
 		return err
 	}
 
-	if _, ok := c.nodes[nodeID]; ok {
-		err = c.nodes[nodeID].clearNodeInfo()
+	if node, ok := c.nodes[nodeID]; ok {
```

Comment:
so far the logic is ok, but it seems that the logic is not achieved for idemptence. 
For example , if removeDmChannelWatchInfosByNodeID success but clearNodeInfo failed, will the task retry to delete nodeId in c.nodes? can the retry success? 

There is usually two way to guarantee idempt:
1) do the io in a batch or trasanction
2) do the critical io in the last step, and make sure all the previous step can be retried with no error.